src/manual/using/actions/ui.md

```diff
@@ -30,24 +30,42 @@ For `action.kind = 'task'`:
     user's Taskcluster credentials. See the next section for some
     important security-related concerns.
 
+For `action.kind = 'hook'`:
+
+-   Rendering the hookPayload using JSON-e with the context describe
+    in the specification.
+-   Calling `Hooks.triggerHook` with the resulting hook identiifers
+    and payload.
+
 User interfaces should ignore actions with unrecognized kinds.
 
-Creating Tasks
---------------
+Security Concerns
+-----------------
 
-When executing an action, a UI must ensure that the user is authorized
-to perform the action, and that the user is not being "tricked" into
-executing an unexpected action.
+When executing an action, a UI must ensure that the user is authorized to
+perform the action, and that the user is not being "tricked" into executing an
+unexpected action. The `actions.json` artifact should be treated as untrusted
+content!
 
 To accomplish the first, the UI should create tasks with the user's
 Taskcluster credentials. Do not use credentials configured as part of
 the service itself!
 
-To accomplish the second, use the decision tasks' `scopes` property as
-the
+### "task" Actions
+
+To accomplish the second for "task" actions, use the decision tasks' `scopes`
+property as the
 [authorizedScopes](https://docs.taskcluster.net/manual/design/apis/hawk/authorized-scopes)
-for the `Queue.createTask` call. This prevents action tasks from doing
-anything the original decision task couldn't do.
+for the `Queue.createTask` call. This prevents action tasks from doing anything
+the original decision task couldn't do.
+
+### "hook" Actions
+
+Hook actions have no such luxury, as they are intended to do things the
+decision task couldn't do.
```

Comment:
We could add that UIs triggering _hook_ actions should validate that the decision task posses the scope:
 * `decision-task:reference-hook-action:<hookGroupId>/<hookId>`

Obviously, the UI would have to call `auth.expandScope(decisionTask.scopes)` to determine the scopes.

This would protect against the case where a try-decision task is tweaked to add an action that triggers the hook to release firefox. As `scm_level_1` doesn't have permission to reference said action.

@djmitche, @imbstack, maybe think this over ^, I'll admit my initial thought is that this is too complicated. That this is something we'll get wrong every time we add a new _hook action_. On the hand scope errors like this should be easy to make visible and understand.

Reply:
Hm, I like that idea.  It does sound like a good bit of work to implement, but not too bad..

Reply:
It is probably worth looking at what the releng [scriptworker](https://github.com/mozilla-releng/scriptworker/blob/master/docs/cot_overview.rst)s are doing. @escapewindow might have some thoughts on this.

Reply:
> It is probably worth looking at what the releng scriptworkers are doing. @escapewindow might have some thoughts on this.

That's a bit of a different situation -- recognizing a "legitimate" decision task for Firefox.  The spec here is more general, and could apply to any project, Firefox or not, with any kind of decision task.